zookeeper-api/src/main/java/org/apache/helix/zookeeper/api/client/MultiOp.java

```diff
@@ -0,0 +1,144 @@
+package org.apache.helix.zookeeper.api.client;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.List;
+
+import org.apache.helix.zookeeper.zkclient.serialize.PathBasedZkSerializer;
+import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.Op;
+import org.apache.zookeeper.ZooDefs;
+import org.apache.zookeeper.data.ACL;
+
+
+/**
+ * A wrapper class of org.apache.zookeeper.Op to support Helix interfaces and monitoring
+ * functionality.
+ */
+public abstract class MultiOp {
+  private final String _path;
+
+  private MultiOp(String path) {
+    _path = path;
+  }
+
+  /**
+   * @return the target path of the operation.
+   */
+  public String getPath() {
+    return _path;
+  }
+
+  /**
+   * Build a new org.apache.zookeeper.Op object for Zookeeper.multi call.
+   * @param serializer
+   * @return org.apache.zookeeper.Op object corresponding to the MultiOp object.
+   */
+  public abstract Op buildZkOp(PathBasedZkSerializer serializer);
```

Comment:
Will ZkSerializer be supported?

Reply:
I don't see any issues with the interfaces.
There is one thing I take as a TODO. If the requested paths are located on different realms, then we won't be able to do it. We may want to check and fail such a request.